### PR TITLE
VC as TH with 1.1

### LINKED ISSUE #1757: mvc/views/view-components.rst "View Components'
- [ ] [View Components as Tag Helpers](https://blogs.msdn.microsoft.com/webdev/2016/11/16/announcing-asp-net-core-1-1/)
- [ ] Update code to use interface/repository pattern

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #1757

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@dpaquette please review

File: aspnetcore/mvc/views/view-components/sample/ViewCompFinal/Startup.cs [hunk 1]
```diff
@@ -21,6 +21,7 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
             {
                 app.UseDeveloperExceptionPage();
             }
+            app.UseDeveloperExceptionPage();
```
Comment (dpaquette) [inline aspnetcore/mvc/views/view-components/sample/ViewCompFinal/Startup.cs]: Did you mean to have this in here twice?

File: aspnetcore/mvc/views/view-components.md [hunk 2]
```diff
@@ -84,9 +84,15 @@ We recommend you name the view file *Default.cshtml* and use the *Views/Shared/C
...
-   ```
+[!code-html[Main](view-components/sample/ViewCompFinal/Views/Todo/IndexFinal.cshtml?range=35)]
+
+## Invoking a view component as a Tag Helper
+
+For ASP.NET Core 1.1 and higher, you can invoke a view component as a [Tag Helper](xref:mvc/views/tag-helpers/intro):
+
+[!code-html[Main](view-components/sample/ViewCompFinal/Views/Todo/IndexTagHelper.cshtml?range=38-39)]
+
+Pascal-cased class and method parameters for tag helpers are translated into their [lower kebab case](http://stackoverflow.com/questions/11273282/whats-the-name-for-dash-separated-case/12273101#12273101). The Tag Helper to invoke a view component is `<vc>`. In the sample above, the `PriorityList` class becomes `priority-list`. The parameters to the view component are passed as arributes in lower kebab case.
```
Comment (dpaquette) [inline aspnetcore/mvc/views/view-components.md]: The wording "The Tag Helper to invoke a view component is `<vc>`" is perhaps a little confusing. This doesn't clearly explain that you prefix the lower-kebab cased view component name with `vc:`. I'm not sure off the top of my head what better wording is but I will come back to this if I think of something. Also, my understanding is that it's not the class name that gets lower kebob-ed, it's the view component name. The name could be different than the class but usually is the class name by convention.

File: aspnetcore/mvc/views/view-components/sample/ViewCompFinal/Views/Todo/Index.cshtml [hunk 3]
```diff
@@ -33,5 +34,7 @@
     }
 </table>
 <div>
-    @await Component.InvokeAsync("PriorityList", new { maxPriority = 2, isDone = false })
+    @*@await Component.InvokeAsync("PriorityList", new { maxPriority = 2, isDone = false })*@
+    <vc:priority-list max-priority="2" is-done="false">
+    </vc:priority-list>
```
Comment (dpaquette) [inline aspnetcore/mvc/views/view-components/sample/ViewCompFinal/Views/Todo/Index.cshtml]: This looks good. Is the old code commented out intentionally? Do we want that there still?

**Comment by Rick-Anderson** [comment]
@dpaquette Great feedback. Not sure I improved it much, maybe you can think of better wording.

**Comment by Rick-Anderson** [comment]
@dpaquette not great, but probably good enough for now.

**Comment by Rick-Anderson** [comment]
@dpaquette  merging for now so I can work on a doc that needs this. Feel free to suggest better wording.

**Pull request merged**